api/geyser/src/main/java/org/geysermc/geyser/api/extension/GeyserExtension.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2019-2022 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.extension;
+
+import org.geysermc.api.GeyserApiBase;
+import java.io.*;
+import java.net.URL;
+import java.net.URLConnection;
+
+/**
+ * This class is to be extended by a Geyser extension
+ */
+public class GeyserExtension {
+    private boolean initialized = false;
+    private boolean enabled = false;
+    private File file = null;
+    private File dataFolder = null;
+    private ClassLoader classLoader = null;
+    private ExtensionLoader loader = null;
+    private ExtensionLogger logger = null;
+    private ExtensionDescription description = null;
+    private GeyserApiBase api = null;
+
+    /**
+     * Called when the extension is loaded
+     */
+    public void onLoad() {
+
+    }
+
+    /**
+     * Called when the extension is enabled
+     */
+    public void onEnable() {
+
+    }
+
+    /**
+     * Called when the extension is disabled
+     */
+    public void onDisable() {
+
+    }
+
+    /**
+     * Gets if the extension is enabled
+     *
+     * @return true if the extension is enabled
+     */
+    public boolean isEnabled() {
+        return this.enabled;
+    }
+
+    /**
+     * Gets if the extension is enabled
+     *
+     * @return true if the extension is enabled
```

Comment:
inaccurate javadoc (no return and modifies state)

Reply:
That was me moving javadocs out of the old Extension class. I must have copied the wrong one without realising.